core/primitives/src/utils/min_heap.rs

```diff
@@ -20,9 +20,19 @@ impl<T: Ord> MinHeap<T> {
         self.inner.peek().map(|Reverse(t)| t)
     }
 
+    // This sadly leaks an abstraction.
+    pub fn peek_mut(&mut self) -> Option<PeekMut<'_, Reverse<T>>> {
```

Comment:
We can define our own `MinHeapPeakMut` to plug the leak. Not sure if it's worth it. Though, if we do wrap the type, we might as well wrap fully?